sdk/identity/azure-identity/azure/identity/_credentials/cli_credential.py

```diff
@@ -0,0 +1,44 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import json
+from datetime import datetime
+from subprocess import run, PIPE
+
+from azure.core.credentials import AccessToken
+from azure.core.exceptions import ClientAuthenticationError
+
+class CliCredential(object):
+
+    _DEFAULT_PREFIX = "/.default"
+
+    def get_token(self, *scopes, **kwargs): # pylint:disable=unused-argument
+        command = ['az', 'account', 'get-access-token']
+
+        if scopes:
+            resource = scopes[0]
+            if resource.endswith(self._DEFAULT_PREFIX):
+                resource = resource[:-len(self._DEFAULT_PREFIX)]
+            
+            command.extend(['--resource', resource])
+
+        # TODO: Add cli installed check
+        get_access_token = run(command, shell=True, stdout=PIPE, stderr=PIPE)
+
+        get_access_token_stdout = get_access_token.stdout.decode("utf-8")
+        get_access_token_stderr = get_access_token.stderr.decode("utf-8")
+
+        if get_access_token_stderr or not get_access_token_stdout:
+            raise ClientAuthenticationError(message=get_access_token_stderr)
+
+        get_access_token_object = json.loads(get_access_token_stdout)
+        access_token = get_access_token_object['accessToken']
+        expires_on = int((
+            datetime.strptime(
+                get_access_token_object['expiresOn'], '%Y-%m-%d %H:%M:%S.%f').utcnow() - datetime(1970, 1, 1)
```

Comment:
utcnow() is class method that return the time as of right now, so actually the object you created with strptime is thrown away and replaced by a new instance with the current time when the method is called:
```python
In [8]: datetime(1970, 1, 1).utcnow()
Out[8]: datetime.datetime(2019, 12, 4, 17, 21, 55, 289927)
```

Reply:
I actually liked the `int(fulltoken['expiresIn'] + time.time())` approach, since durations don't have timezone troubles. Otherwise you need to parse this guy with "timezone.utc", that doesn't exist in 2.7, so you'll need a hack like [that](https://github.com/Azure/msrest-for-python/blob/b505e3627b547bd8fdc38327e86c70bdb16df061/msrest/serialization.py#L66-L86), and makes the code more subtle IMHO.